### PR TITLE
Allow complex queries with table joins

### PR DESCRIPTION
**Pull request by xjunior**
Allow using external columns for complex joins. In the bellow example we're able to query the user membership within a group:

```ruby
  class User < ApplicationRecord
    has_and_belongs_to_many :groups

    def self.scim_queryable_attributes
      return {
        'groups'       => { column: Group.arel_table[:id] },
        'groups.value' => { column: Group.arel_table[:id] },
      }
    end
  end
```

Then:

```ruby
  class UsersController < Scimitar::ActiveRecordBackedResourcesController
    def storage_class
      ::User
    end

    def storage_scope
      ::User.left_joins(:groups)
    end
  end
```

### CODE REVIEW

**Review by wadewinningham — APPROVED** [review]

**Comment by pond** [comment]
Sorry to get to this so late.

The code looks good and I'm very enthusiastic about your test coverage. The only missing thing is documentation - sorry, can be tedious, but it's super-important. New features need clear and comprehensive documentation in `README.md`, otherwise nobody knows the feature exists and future maintainers risk stumbling across code with no apparent purpose.

It looks like a subsection addition under `### Controllers` at (v2.4.3) line 224. The important part is ideally not to just give the example in the PR description, but to explain what's really happening. We know, because we know how the under-hood implementation works. Regular gem users won't.

* You have a User join in `storage_scope`
* You have a `scim_queryable_attributes` with `Group.arel_table[:id]`
* Please make sure the reader understands in particular why there's the magic `arel_table[:id]` there.

The reader wants to understand what this feature is for and how it would work _for them_ with their own data model, so that's why it's not good to force the reader to rely on assumptions and guesswork (e.g. be very clear that you have a run-of-the-mill ApplicationRecord-based User model with a `belong_to` - or is it HABTM? - relationship to another run-of-the-mill ApplicationRecord-based Group model).

Hope that's OK.

File: README.md
Comment (pond) [inline README.md:44]: FYI those were deliberate as it's much easier to skim through the source code of Markdown and see where major headings are if there's a visual break. The file ought to be consistent with that style of line breaking before L2 headings (and L3 on a judgement call depending on context and surrounding information density).

This won't halt you PR but I'll be manually restoring that once it's merged in a subsequent commit.

Comment (xjunior) [reply README.md:44]: @pond I'm sorry, this was a markdown linter. I'll revert the undesired changes.

File: README.md
Comment (pond) [inline README.md:206]: It's good to include something in an example but in this case it's quite an advanced jump and has no explanation of why it's there or what it means, especially without a clear idea of MockGroup. Note the preamble at line 115, which points to the RDoc format (note RDoc, not Markdown) comments in the source code at

https://github.com/RIPAGlobal/scimitar/blob/main/app/models/scimitar/resources/mixin.rb

...so perhaps that's the place to explain why and how AREL table references can be used?

File: README.md
Comment (pond) [inline README.md:270]: Very useful addition, thanks.

**Review by pond — CHANGES_REQUESTED** [review]
Again is all good except docs - AREL use still not explained - see review comment for a suggestion about putting that into the RDoc comments of the mixin.

**Comment by xjunior** [comment]
@pond let me know how you feel about the changes. I expanded the options documentation a bit, allowing us to introduce more queryable attribute options in the future (I can see a `case_sensitive` option, for instance).

**Review by pond — APPROVED** [review]
The comment additions are enough. The options docs aren't really RDoc, but I'll clean that up before releasing the bumped gem version. This is good to go!

**Comment by pond** [comment]
Formatting & RDoc markup tweaks can be seen in the diff at https://github.com/RIPAGlobal/scimitar/pull/67/files - hope you're OK with those changes; let me know if not (e.g. comment on that PR, even if it's merged).

**Pull request merged**
